src/test/run-pass/env-vars-iter.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::process::Command;
+use std::env;
+
+fn check_var(var: Option<(String, String)>) {
+    if let Some((key, val)) = var {
+        match &*key {
+            "FOO" => assert_eq!(val, "BAR"),
+            "BAR" => assert_eq!(val, "BAZ"),
+            "BAZ" => assert_eq!(val, "FOO"),
+            _ => panic!("Unexpected environment variable {}={}", key, val),
+        }
+    } else {
+        panic!("Expected an environment variable, got None")
+    }
+}
+
+fn main() {
+    if let Some(arg) = env::args().nth(1) {
+        match &*arg {
+            "empty" => {
+                assert_eq!(env::vars().count(), 0);
+                assert_eq!(env::vars().next(), None);
+                assert_eq!(env::vars().nth(1), None);
+                assert_eq!(env::vars().last(), None);
+            },
+            "many" => {
+                assert_eq!(env::vars().count(), 3);
+                check_var(env::vars().nth(1));
+                check_var(env::vars().last());
+                let mut vars = env::vars();
+                check_var(vars.next());
+                check_var(vars.next());
+                check_var(vars.next());
+                assert_eq!(vars.next(), None);
+            },
+            arg => {
+                panic!("Unexpected arg {}", arg)
+            },
+        }
+    } else {
+      let status = Command::new(env::current_exe().unwrap())
+                           .arg("empty")
+                           .env_clear()
```

Comment:
This may or may not pass on Windows due to https://github.com/rust-lang/rust/issues/31259